pkg/server/handler.go

```diff
@@ -91,6 +94,24 @@ func WithClusterScope(apiHandler http.Handler) http.HandlerFunc {
 		} else {
 			clusterName = req.Header.Get("X-Kubernetes-Cluster")
 		}
+		if clusterName == "" {
+			// let's get the information from the request
+			resp, _, err := authorization.Authenticator.AuthenticateRequest(req)
+			if err != nil {
+				responsewriters.ErrorNegotiated(
+					apierrors.NewInternalError(err),
+					errorCodecs, schema.GroupVersion{},
+					w, req)
+				return
+			}
+			if resp != nil {
+				extraInfo := resp.User.GetExtra()
+				if val, ok := extraInfo[serviceaccount.ClusterNameKey]; ok {
+					klog.Infof("Set cluster name from request to: %q", val[0])
+					clusterName = val[0]
+				}
+			}
+		}
```

Comment:
this is the wrong spot. We have to make the service account on the pcluster side to talk to the right server URL.